source/common/tls/server_context_impl.cc

```diff
@@ -42,18 +42,24 @@
 namespace Envoy {
 namespace {
 
-bool cbsContainsU16(CBS& cbs, uint16_t n) {
+Ssl::CurveNIDSupportedVector getClientCurveNIDSupported(CBS& cbs) {
+  Ssl::CurveNIDSupportedVector cnsv{};
   while (CBS_len(&cbs) > 0) {
     uint16_t v;
     if (!CBS_get_u16(&cbs, &v)) {
-      return false;
+      return cnsv;
     }
-    if (v == n) {
-      return true;
+    if (v == SSL_SIGN_ECDSA_SECP256R1_SHA256 || v == SSL_CURVE_SECP256R1) {
```

Comment:
If I'm understanding this correctly, the two values in each comparison are for TLS 1.2 and 1.3, correct? Can you add a comment explaining that here?